From 3c30ddddb1329ee0893aea3b2f2b02741542993c Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Wed, 29 Feb 2012 14:51:13 -0500
Subject: [PATCH] Fix _sync_power_states to obtain correct 'state'.

Update the compute manager to obtain state from the get_info
dict correctly.

Fixes 'dict' object has no attribute 'state' error that occur
when syncing power states.

Fixes LP Bug #934276.

Change-Id: I3e3602018f4c38f78b48186bfca78eda362a9e54
---
 nova/compute/manager.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/nova/compute/manager.py b/nova/compute/manager.py
index 2221946..3df5852 100644
--- a/nova/compute/manager.py
+++ b/nova/compute/manager.py
@@ -2242,7 +2242,7 @@ class ComputeManager(manager.SchedulerDependentManager):
             db_power_state = db_instance['power_state']
             try:
                 vm_instance = self.driver.get_info(db_instance)
-                vm_power_state = vm_instance.state
+                vm_power_state = vm_instance['state']
             except exception.InstanceNotFound:
                 LOG.warn(_("Instance found in database but not known by "
                            "hypervisor. Setting power state to NOSTATE"),
